### DISCUSSION TITLE
How Long did `pyvista` Take to Make?

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvista#3181] [Q&A]
As a developer, I'm curious to know how long it to for `pyvista` to first come to fruition and where it all started. `pyvista` is constantly developing, so it's probably not fair to say "how long" since that implies "it's done". I'm curious about how long it took from the start of the repo to the first usable release (which, again, may be up to interpetation).

What is `pyvista`'s history?

**Comment by tkoyama010** [comment]
You can see a brief history [here](https://tutorial.pyvista.org/tutorial/00_intro/index.html#brief-history). For more details on the background, please ask @akaszynski and @banesullivan :wink:

**Reply by banesullivan** [reply]
The contributor's graph is actually a really fantastic way to gather some insights for this question: https://github.com/pyvista/pyvista/graphs/contributors

![Screen Shot 2022-08-29 at 9 08 15 AM](https://user-images.githubusercontent.com/22067021/187233446-7c0bb1d7-50d2-468f-87ca-7c3eaee27648.png)


I'd say it took about ~3 years with many contributors to get PyVista to be as widely used as it is today - meaning to get it to a stable and useable state. However, as you mention, this is up for interpretation... Alex and I have put a lot of blood, sweat, and tears into PyVista over the last ~5 years, with countless late nights, so saying it tool X many labor hours to make PyVista is probably impossible to guess at this point.

PyVista is nowhere near done either, and I think we're about to start a new phase of development soon here that will really ramp things up 😉  as there is much to be done to bring VTK and PyVista closer together and to improve the user experience in Jupyter.

**Reply by adam-grant-hendry** [reply]
Wow, that's some incredible work!

...What happened in 2019? HUGE spike there!

**Reply by banesullivan** [reply]
> What happened in 2019? HUGE spike there!

That's when I joined the project 😉
